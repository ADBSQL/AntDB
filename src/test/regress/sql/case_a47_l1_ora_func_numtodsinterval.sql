set grammar to oracle;
set datestyle='ISO,YMD';
select to_date('2015-2-5 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY')  as ndate from dual;
select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY')  as ndate from dual;
select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'HOUR')  as ndate from dual;
select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'minute')  as ndate from dual;
select to_date('2015-2-28 23:59:59','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(1,'second')  as ndate from dual;
select to_date('2015-2-28 23:59:59','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('1','second')  as ndate from dual;
select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.567,'second')  as ndate from dual;
select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.123456789,'day')  as ndate from dual;
select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.123456789,'hour')  as ndate from dual;
select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('3.123456789','hour')  as ndate from dual;
select to_timestamp('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('3.123456789','hour')  as ndate from dual;
select to_char(numtodsinterval('3.123456789','hour'),'hh24:mi:ss')  as ndate from dual;
select to_char(numtodsinterval('3.1','hour') + numtodsinterval('2.1','minute'),'hh24:mi:ss') as ndate from dual;
select numtodsinterval('3.123456789','')  as ndate from dual;
select numtodsinterval('3.1',null) + numtodsinterval('2.1','minute') as ndate from dual;
select to_char(numtodsinterval('3.1','minute') * 3,'hh24:mi:ss') as ndate from dual;


create table t4test(id int, dt date);
insert into t4test values(1,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'second'));
insert into t4test values(2,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'minute'));
insert into t4test values(3,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'HOUR'));
insert into t4test values(4,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY'));

select * from t4test order by id;
update t4test set dt=dt+numtodsinterval(24,'HOUR');
select * from t4test order by id;

drop table t4test;
