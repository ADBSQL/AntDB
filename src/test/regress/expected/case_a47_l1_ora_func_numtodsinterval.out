set grammar to oracle;
set datestyle='ISO,YMD';
select to_date('2015-2-5 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY')  as ndate from dual;
        NDATE        
---------------------
 2015-02-15 10:55:55
(1 row)

select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY')  as ndate from dual;
        NDATE        
---------------------
 2015-03-07 10:55:55
(1 row)

select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'HOUR')  as ndate from dual;
        NDATE        
---------------------
 2015-02-25 20:55:55
(1 row)

select to_date('2015-2-25 10:55:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'minute')  as ndate from dual;
        NDATE        
---------------------
 2015-02-25 11:05:55
(1 row)

select to_date('2015-2-28 23:59:59','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(1,'second')  as ndate from dual;
        NDATE        
---------------------
 2015-03-01 00:00:00
(1 row)

select to_date('2015-2-28 23:59:59','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('1','second')  as ndate from dual;
        NDATE        
---------------------
 2015-03-01 00:00:00
(1 row)

select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.567,'second')  as ndate from dual;
          NDATE          
-------------------------
 2015-02-28 02:01:04.567
(1 row)

select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.123456789,'day')  as ndate from dual;
           NDATE           
---------------------------
 2015-03-03 04:58:47.66657
(1 row)

select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(3.123456789,'hour')  as ndate from dual;
           NDATE           
---------------------------
 2015-02-28 05:08:25.44444
(1 row)

select to_date('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('3.123456789','hour')  as ndate from dual;
           NDATE           
---------------------------
 2015-02-28 05:08:25.44444
(1 row)

select to_timestamp('2015-2-28 2:1:1','YYYY-MM-DD hh24:mi:ss') + numtodsinterval('3.123456789','hour')  as ndate from dual;
           NDATE           
---------------------------
 2015-02-28 05:08:25.44444
(1 row)

select to_char(numtodsinterval('3.123456789','hour'),'hh24:mi:ss')  as ndate from dual;
  NDATE   
----------
 03:07:24
(1 row)

select to_char(numtodsinterval('3.1','hour') + numtodsinterval('2.1','minute'),'hh24:mi:ss') as ndate from dual;
  NDATE   
----------
 03:08:06
(1 row)

select numtodsinterval('3.123456789','')  as ndate from dual;
 NDATE 
-------
 
(1 row)

select numtodsinterval('3.1',null) + numtodsinterval('2.1','minute') as ndate from dual;
 NDATE 
-------
 
(1 row)

select to_char(numtodsinterval('3.1','minute') * 3,'hh24:mi:ss') as ndate from dual;
  NDATE   
----------
 00:09:18
(1 row)

create table t4test(id int, dt date);
insert into t4test values(1,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'second'));
insert into t4test values(2,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'minute'));
insert into t4test values(3,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'HOUR'));
insert into t4test values(4,to_date('2015-2-5 10:45:55','YYYY-MM-DD hh24:mi:ss') + numtodsinterval(10,'DAY'));
select * from t4test order by id;
 ID |         DT          
----+---------------------
  1 | 2015-02-05 10:46:05
  2 | 2015-02-05 10:55:55
  3 | 2015-02-05 20:45:55
  4 | 2015-02-15 10:45:55
(4 rows)

update t4test set dt=dt+numtodsinterval(24,'HOUR');
select * from t4test order by id;
 ID |         DT          
----+---------------------
  1 | 2015-02-06 10:46:05
  2 | 2015-02-06 10:55:55
  3 | 2015-02-06 20:45:55
  4 | 2015-02-16 10:45:55
(4 rows)

drop table t4test;
